hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -82,7 +88,20 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         requireNonNull(context);
         final var op = context.body().cryptoCreateAccountOrThrow();
         pureChecks(context.body());
+
         if (op.hasKey()) {
+            final var key = op.key();
+            if (op.alias() != null && !op.alias().equals(Bytes.EMPTY)) {
+                final var alias = op.alias();
+                // add evm address key to req keys only if it is derived from a key, diff than the admin key
+                final var isAliasDerivedFromDiffKey = !key.hasEcdsaSecp256k1()
+                        || !Arrays.equals(
+                                recoverAddressFromPubKey(key.ecdsaSecp256k1().toByteArray()), alias.toByteArray());
```

Comment:
This might be easier to read with a single inversion:
```suggestion
                final var isAliasDerivedFromDiffKey = !(key.hasEcdsaSecp256k1()
                        && Arrays.equals(
                                recoverAddressFromPubKey(key.ecdsaSecp256k1().toByteArray()), alias.toByteArray()));
```

Reply:
Fixed. Thanks